extensions/objects/templates/literally-canvas-diagram-editor.component.spec.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the literally canvas diagram editor.
+ */
+
+describe('LiterallyCanvasDiagramEditor', function() {
+  var LCDiagramEditorCtrl = null;
+  var mockLiterallyCanvas = {
+    setImageSize: function(width, height) {
+      var text = (
+        'The updated diagram width is ' + width +
+        ' and height is ' + height);
+      return text;
+    }
+  };
+
+  beforeEach(angular.mock.module('oppia'));
+
+  beforeEach(angular.mock.inject(function($componentController) {
+    LCDiagramEditorCtrl = $componentController('literallyCanvasDiagramEditor');
+    var mockDocument = document.createElement('div');
+    mockDocument.setAttribute('id', LCDiagramEditorCtrl.lcID);
+    document.getElementById = jasmine.createSpy(
+      'HTML element').and.returnValue(mockDocument);
```

Comment:
```suggestion
    spyOn(document, 'getElementById').and.returnValue(mockDocument);
```

Reply:
This doesn't seem to working as expected because document.getElementById is used inside angular.element(document).ready function. Another alternate solution is to use 
```
var mockDocument = document.createElement('div');
mockDocument.setAttribute('id', LCDiagramEditorCtrl.lcID);
var $document = angular.element(document);
$document.find('body').append(mockDocument.outerHTML);
```

Reply:
Ok, that's fine!